src/viewport/s3_utils.py

```
@@ -197,3 +198,61 @@ def generate_thumbnail_object_key(original_object_key: str) -> str:
+
+    Args:
+        original_object_key: Original object key (e.g., 'gallery_id/photo_id.jpg')
+
+    Returns:
+        Poster object key in AVIF format (e.g., 'gallery_id/photo_id_poster.avif')
+    """
+    if "/" in original_object_key:
+        gallery_id, filename = original_object_key.split("/", 1)
+        photo_id = filename.rsplit(".", 1)[0] if "." in filename else filename
+        return f"{gallery_id}/{photo_id}_poster.avif"
+    else:
```

Ruff (0.15.20)

[warning] Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)